### PR TITLE
RA VPN profile creation sample

### PR DESCRIPTION
**Pull request by berezovskyi-oleksandr**
Added RA VPN profile creation
Added SmartAgent connection registration for EVALUATED FTDs
Added RA VPN license enabling
Added anyconnect package file downloading and registration

Timings on my machine for 10 parallel connections on 10 FTDs:
1min 19sec - re-run on fully-deployed environment
1min 57sec - re-run with 1 clean FTD
2min 17sec - re-run with 10 clean FTDs

### CODE REVIEW

File: samples/ftd_configuration/anyconnect_package_file.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+- hosts: all
+  connection: httpapi
+  gather_facts: no
+  vars_files:
+  - variables.yaml
+  tasks:
+  - name: Check Anyconnect Package File presense
```
Comment (annikulin) [inline samples/ftd_configuration/anyconnect_package_file.yaml]: Let's use `AnyConnect` case to match the product name...

File: samples/ftd_configuration/anyconnect_package_file.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  - name: Add Anyconnect Package File
+    ftd_configuration:
+      operation: addAnyConnectPackageFile
+      data:
+        name: anyconnect_package_file
+        platformType: LINUX
+        diskFileName: '{{ anyconnect_linux_image_file_on_device.fileName }}'
+        type: anyconnectpackagefile
+      register_as: anyconnect_package_file
+    when: 'anyconnect_linux_image_file_on_device is defined'
```
Comment (annikulin) [inline samples/ftd_configuration/anyconnect_package_file.yaml]: Missing empty line at the end.

File: samples/ftd_configuration/anyconnect_package_file.yaml (hunk 1 above)
Comment (annikulin) [inline samples/ftd_configuration/anyconnect_package_file.yaml]:
```suggestion
  - name: Check Anyconnect Package File presence
```

File: samples/ftd_configuration/anyconnect_package_file.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  gather_facts: no
+  vars_files:
+  - variables.yaml
+  tasks:
+  - name: Check Anyconnect Package File presense
+    ftd_configuration:
+      operation: getAnyConnectPackageFileList
+      register_as: all_anyconnect_package_files
+
+  - name: Check Anyconnect Image presense
```
Comment (annikulin) [inline samples/ftd_configuration/anyconnect_package_file.yaml]: Maybe `Check AnyConnect file presence on local filesystem`? Just to clarify that we are checking the file on local filesystem, not on the FTD.

File: samples/ftd_configuration/ra_vpn.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
+- name: Setup SmartAgent
+  import_playbook: smart_agent_connection.yaml
+
+- name: Setup required license
+  import_playbook: ra_vpn_license.yaml
+
+- name: Setup anyconnect vpn files
```
Comment (annikulin) [inline samples/ftd_configuration/ra_vpn.yaml]:
```suggestion
- name: Setup AnyConnect VPN package files
```

File: samples/ftd_configuration/ra_vpn.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        name: NGFW-Remote-Access-VPN # NOTE: this name is ignored on backend
+        vpnGatewaySettings:
+          outsideInterface: '{{ all_phisical_interfaces | selectattr("name", "eq", "outside") | list | first }}'
+          serverCertificate: '{{ all_internal_certificates | first }}'
+          exemptNatRule: false
+          bypassAccessControlForVPNTraffic: false
+          clientCACertificates: []
+          insideInterfaces: []
+          insideNetworks: []
+          outsideFqdn: ""
```
Comment (annikulin) [inline samples/ftd_configuration/ra_vpn.yaml]: Let's put some value here. E.g., '{{inventory_hostname}}.cisco.com'

File: samples/ftd_configuration/smart_agent_connection.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
+- hosts: all
```
Comment (annikulin) [inline samples/ftd_configuration/smart_agent_connection.yaml]: Let's merge this playbook with `smart_license.yml` as they are essentially doing the same things.

File: samples/ftd_configuration/anyconnect_package_file.yaml (hunk 2 above)
Comment (berezovskyi-oleksandr) [reply samples/ftd_configuration/anyconnect_package_file.yaml]: Done

File: samples/ftd_configuration/anyconnect_package_file.yaml (hunk 1 above)
Comment (berezovskyi-oleksandr) [reply samples/ftd_configuration/anyconnect_package_file.yaml]: Done

Comment (berezovskyi-oleksandr) [reply samples/ftd_configuration/anyconnect_package_file.yaml]: Done

File: samples/ftd_configuration/anyconnect_package_file.yaml (hunk 3 above)
Comment (berezovskyi-oleksandr) [reply samples/ftd_configuration/anyconnect_package_file.yaml]: Done

File: samples/ftd_configuration/ra_vpn.yaml (hunk 4 above)
Comment (berezovskyi-oleksandr) [reply samples/ftd_configuration/ra_vpn.yaml]: Done

File: samples/ftd_configuration/ra_vpn.yaml (hunk 5 above)
Comment (berezovskyi-oleksandr) [reply samples/ftd_configuration/ra_vpn.yaml]: Done

File: samples/ftd_configuration/smart_agent_connection.yaml (hunk 6 above)
Comment (berezovskyi-oleksandr) [reply samples/ftd_configuration/smart_agent_connection.yaml]: Done

**Pull request merged**